### ISSUE TITLE
x/build/cmd/tip: remove non-functional talks.golang.org server and deploy instructions

### ISSUE THREAD

**Issue by dmitshur** [golang/go#36825]
I've recently realized that, due to a series of unfortunate events spanning 2014-2020, we have accidentally ended up with 2 sets of deploy instructions for [talks.golang.org](https://talks.golang.org):

1. [golang.org/x/build/cmd/tip README](https://github.com/golang/build/blob/c373c47710ad8ce94647ed9cfb258414ad995467/cmd/tip/README#L1-L18)
2. [golang.org/x/talks README](https://github.com/golang/talks/blob/083066ceb3152d883ec6a8ee67e00fd88330fc0c/README#L7-L14)

We should pick one as canonical (likely the one in `x/build/cmd/tip`, because it has an auto-deploy feature), and update READMEs in both places to avoid any future confusion.

Making this issue quickly so I don't forget; will add more history/background later.

/cc @cagedmantis @toothrot @andybons

**Comment by rsc** [comment]
Definitely agree about not having two sets of instructions.

It's unfortunate that the x/build version requires Docker,
but if it happens automatically on push so that mere mortals
never have to run it, that's fine.

**Comment by dmitshur** [comment]
I investigated whether the deploy instructions in x/build/cmd/tip (which have the auto-deploy feature) can still be used today, but they have atrophied and cannot be used without making changes to get them functional again. I tried making small and safe changes to get them working, but it turned out larger changes are required to get them functional without also breaking tip.golang.org in the process. (I've described how they're broken in a separate issue #38120.)

The deploy instructions in x/talks are functional and have been used most recently. They lack the auto-deploy feature, unfortunately.

We could resolve this issue by continuing to use deploy instructions in golang.org/x/talks README for now, delete the non-functional instructions in x/build/cmd/tip (or at least mark them as "broken" and in need of fixing), and file a tracking issue about restoring the auto-deploy of x/talks content feature that used to exist. It can be resolved by making x/build/cmd/tip functional for serving talks.golang.org again, or another way.

Alternatively, we can fix x/cmd/tip deploy instructions (i.e., resolve #38120) and use them. I'll investigate that issue some more first.

**Comment by rsc** [comment]
Now that we have //go:embed I believe the right fix here is to use //go:embed so that no files have to be copied across repos as part of a deploy.

**Comment by dmitshur** [comment]
> The deploy instructions in x/talks are functional and have been used most recently.

As we found out from a deploy attempt for #43455, they stopped being functional. I opened issue #44164 and sent [CL 290192](https://golang.org/cl/290192) to fix it. (As part of that work, I also revisited this issue; see below for updates.)

> Now that we have //go:embed I believe the right fix here is to use //go:embed so that no files have to be copied across repos as part of a deploy.

The `cmd/present` code lives in x/tools, while the talk content is in x/talks, so it seems some copying is unavoidable. I considered moving the App Engine-specific code from x/tools/cmd/present to x/talks (similarly to what we've done in the x/blog repo) to get around having to copy code, but that resulted in much code duplication.

> Alternatively, we can fix x/cmd/tip deploy instructions (i.e., resolve #38120) and use them. I'll investigate that issue some more first.

I investigated that option and pretty much got it working in a staging instance. However, I decided not to continue with that approach for reasons described in https://github.com/golang/go/issues/38120#issuecomment-775365482.

So, I'll update this issue to be about removing the non-functional x/build/cmd/tip deploy for talks, and we can continue to use the simpler deployment in x/talks README instead (which will be functional as soon as #44164 is resolved). More can be done later as part of future work.
